executor.go

```diff
@@ -56,25 +56,22 @@ func (x *executor) Execute(req *Request, re ResponseEmitter, env Environment) er
 			postRunCh = make(chan error)
 		)
 
-		if postRun == nil {
-			close(postRunCh)
-			return postRunCh
-		}
-
-		// check if we have a formatter for this emitter type
+		// Check if we have a formatter for this emitter type.
 		typer, isTyper := re.(interface {
 			Type() PostRunType
 		})
-		if isTyper &&
-			formatters[typer.Type()] != nil {
+		if isTyper {
 			postRun = formatters[typer.Type()]
-		} else {
+		}
+		// If not, just return nil via closing.
+		if postRun == nil {
 			close(postRunCh)
 			return postRunCh
 		}
 
-		// redirect emitter to us
-		// and start waiting for emissions
+		// Otherwise, relay emitter responses
+		// from Run to PostRun, and
+		// from PostRun to the original emitter.
```

Comment:
I think this makes sense, but that's because I wrote it.
If it could be reworded or is confusing, someone let me know.